cli_tools/gce_ovf_export/exporter/flags.go

```diff
@@ -0,0 +1,78 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package ovfexporter
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/flags"
+	ovfexportdomain "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_export/domain"
+)
+
+// RegisterFlags registers OVF exporter CLI flags with args
+func RegisterFlags(ovfExportArgs *ovfexportdomain.OVFExportArgs, args []string) error {
+	flagSet := flag.NewFlagSet("ovf-export", flag.ContinueOnError)
+	// Don't write parse errors to stdout, instead propagate them via an
+	// exception since we use flag.ContinueOnError.
+	flagSet.SetOutput(ioutil.Discard)
+
+	flagSet.Var((*flags.TrimmedString)(&ovfExportArgs.InstanceName), ovfexportdomain.InstanceNameFlagKey,
+		"Name of the VM Instance to be exported.")
+	flagSet.Var((*flags.TrimmedString)(&ovfExportArgs.MachineImageName), ovfexportdomain.MachineImageNameFlagKey,
+		"Name of the Google machine image to be exported.")
+	flagSet.Var((*flags.LowerTrimmedString)(&ovfExportArgs.ClientID), ovfexportdomain.ClientIDFlagKey,
+		"Identifies the client of the exporter, e.g. `gcloud` or `pantheon`")
+	flagSet.Var((*flags.TrimmedString)(&ovfExportArgs.ClientVersion), "client-version",
+		"Identifies the version of the client of the exporter.")
+	flagSet.Var((*flags.TrimmedString)(&ovfExportArgs.DestinationURI), ovfexportdomain.DestinationURIFlagKey,
+		"Google Cloud Storage URI of the OVF or OVA file to export to. For example: gs://my-bucket/my-vm.ovf.")
+	flagSet.Var((*flags.LowerTrimmedString)(&ovfExportArgs.OvfFormat), ovfexportdomain.OvfFormatFlagKey,
+		"One of: `ovf` or `ova`. Defaults to `ovf`. If `ova` is specified, exported OVF package will be packed as an OVA archive and individual files will be removed from GCS.")
+	flagSet.Var((*flags.LowerTrimmedString)(&ovfExportArgs.DiskExportFormat), "disk-export-format",
+		"format for disks in OVF, such as vmdk, vhdx, vpc, or qcow2. Any format supported by qemu-img is supported by OVF export. Defaults to `vmdk`.")
```

Comment:
Thoughts on limiting to vmdk at launch, and then seeing whether there's demand for more? The reason: If we implemented native vmdk export in the future, then we could make a clean transition without having a parallel track for the other formats.

Might be moot, though, since we already have the other formats for image export.

Reply:
I'm not sure what is the distribution of supported formats in environments where exported OVFs will be imported. It's is likely supported everywhere but I don't have that data. With this idea, we're basically trading lower complexity of our solution with potential compatibility issues. As you mentioned, image export already officially supports all these formats, and OVF export uses image export under the hood. So there's not much to gain in simplifying OVF export.